posthog/auth.py

```diff
@@ -83,14 +84,9 @@ def authenticate(cls, request: Union[HttpRequest, Request]) -> Optional[Tuple[An
         key_last_used_at = personal_api_key_object.last_used_at
         # Only updating last_used_at if the hour's changed
         # This is to avooid excessive UPDATE queries, while still presenting accurate (down to the hour) info in the UI
-        if key_last_used_at is None or (now.year, now.month, now.day, now.hour) > (
-            key_last_used_at.year,
-            key_last_used_at.month,
-            key_last_used_at.day,
-            key_last_used_at.hour,
-        ):
```

Comment:
About https://github.com/PostHog/posthog/pull/10383#issuecomment-1422965994, I don't think there was a good reason for comparing this way instead of using a `timedelta`, but I'm also surprised that this wasn't working as intended. In what situation was the `>` true when it shouldn't it?

Reply:
problem wasn't this comparison, but in the next line, where we weren't updating the token

Reply:
changing this as well so all saves don't spike at the first minute of the hour